### PR TITLE
Ports/psoc6 RTOS scan example cleanup

### PR DESCRIPTION
**Pull request by jaenrig-ifx**
* Removed all scan example files and functions which should be transferred to the module. 

We won´t merge until the network module at least provides the scanning function.

### CODE REVIEW

**Review by boramonideep — APPROVED** [review]
I dont understand the specifics but looks good.

File: extmod/modlwip.c [hunk 1]
```diff
@@ -1620,13 +1620,18 @@ STATIC MP_DEFINE_CONST_OBJ_TYPE(
 // Support functions for memory protection. lwIP has its own memory management
 // routines for its internal structures, and since they might be called in
 // interrupt handlers, they need some protection.
+
+// Added protection flag. Only few ports are providing and OS
```
Comment (bargfred) [inline extmod/modlwip.c:1624]: I think the comment should be rephrased.

File: ports/psoc6/main.c [hunk 2]
```diff
@@ -114,84 +97,82 @@ int main(int argc, char **argv) {
...
 
+soft_reset:
+
+    mp_init();
 
-    result = cyhal_gpio_init(CYBSP_USER_LED, CYHAL_GPIO_DIR_OUTPUT, CYHAL_GPIO_DRIVE_STRONG, CYBSP_LED_STATE_OFF);
-    error_handler(result, NULL);
-    is_led_initialized = true;
+    // ANSI ESC sequence for clear screen. Refer to  https://stackoverflow.com/questions/517970/how-to-clear-the-interpreter-console
+    // mp_printf(&mp_plat_print, "\033[H\033[2J");
```
Comment (bargfred) [inline ports/psoc6/main.c:115]: This should be commented in again now that the test phase is over.

File: ports/psoc6/main.c [hunk 3]
```diff
@@ -36,70 +36,53 @@
...
-};
+    xTaskCreate(mpy_task, "MicroPython task", MPY_TASK_STACK_SIZE, NULL, MPY_TASK_PRIORITY, &mpy_task_handle);
+    vTaskStartScheduler();
 
+    // Should never get here
+    CY_ASSERT(0);
+    return 0;
+}
 
 #if MICROPY_ENABLE_GC
```
Comment (bargfred) [inline ports/psoc6/main.c:67]: I would move the following defs and extern statements to the top before main to stick to common practice.

**Review by bargfred — APPROVED** [review]
Looks good !

File: extmod/modlwip.c (hunk 1 above)
Comment (jaenrig-ifx) [reply extmod/modlwip.c]: done

File: ports/psoc6/main.c (hunk 3 above)
Comment (jaenrig-ifx) [reply ports/psoc6/main.c]: done

File: ports/psoc6/main.c (hunk 2 above)
Comment (jaenrig-ifx) [reply ports/psoc6/main.c]: done

File: ports/psoc6/modules/network/network_ifx_whd.c [hunk 4]
```diff
@@ -34,38 +34,41 @@
...
 #include "lwip/netif.h"
+#include "lwip/dns.h"
 #include "whd.h"
 #include "whd_network_types.h"
 #include "whd_wlioctl.h"
 #include "network_ifx_whd.h"
 #include "whd_wifi_api.h"
 #include "extmod/modnetwork.h"
-#include "cy_wcm.h"
+#include "cy_wcm.c"
```
Comment (jaenrig-ifx) [inline ports/psoc6/modules/network/network_ifx_whd.c:44]: I will bet this inclusion of .c is due to some static or private scope variables or objects of the cy_wcm.c?

**Pull request merged**